kubeflow/trainer/backends/container/utils.py

```
@@ -100,12 +105,21 @@ def build_pip_install_cmd(trainer: types.CustomTrainer) -> str:
         return ""
 
     index_urls = trainer.pip_index_urls or list(constants.DEFAULT_PIP_INDEX_URLS)
-    main_idx = index_urls[0]
-    extras = " ".join(f"--extra-index-url {u}" for u in index_urls[1:])
-    quoted = " ".join(f'"{p}"' for p in pkgs)
+    main_idx = shlex.quote(index_urls[0])
+    extras = " ".join(f"--extra-index-url {shlex.quote(u)}" for u in index_urls[1:])
+    quoted = " ".join(shlex.quote(p) for p in pkgs)
+    pip_env = "PIP_DISABLE_PIP_VERSION_CHECK=1 PIP_BREAK_SYSTEM_PACKAGES=1"
+    pip_base = f"python -m pip install --no-warn-script-location --index-url {main_idx} {extras}"
+    log_file = "/tmp/pip_install.log"
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/pip_install.log"

(S108)